rct229/rules/section5.py

```diff
@@ -97,3 +99,88 @@ def rule_check(self, context, calc_vals, data=None):
 
 
 # ------------------------
+
+
+class Section5Rule44(RuleDefinitionListIndexedBase):
+    """Rule 44 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule44, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section5Rule44.BuildingRule(),
+            index_rmr="baseline",
+            id="5-44",
+            description="The infiltration modeling method in the baseline includes adjustment for weather and building operation.",
+            rmr_context="buildings",
+        )
+
+    class BuildingRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule44.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={"$..infiltration[*]": ["modeling_method"]},
+            )
+
+        def get_calc_vals(self, context, data=None):
+            baseline_infiltration = find_all("$..infiltration[*]", context.baseline)
+            failing_infiltration_ids = [
+                b_infiltration["id"]
+                for b_infiltration in baseline_infiltration
+                if b_infiltration["modeling_method"] != CONSTANT
+            ]
+            return {"failing_infiltration_ids": failing_infiltration_ids}
+
+        def rule_check(self, context, calc_vals, data=None):
+            return len(calc_vals["failing_infiltration_ids"]) == 0
+
+
+# ------------------------
+
+
+class Section5Rule46(RuleDefinitionListIndexedBase):
+    """Rule 44 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule46, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section5Rule46.BuildingRule(),
+            index_rmr="proposed",
+            id="5-46",
+            description="The infiltration shall be modeled using the same methodology and adjustments for weather and building operation in both the proposed design and the baseline building design.",
+            rmr_context="buildings",
+        )
+
+    class BuildingRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule46.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True)
```

Comment:
Should there be required fields? 
1) all zones have an `infiltration` field
2) all infiltration objects have `algorithm_name` and `model_method fields`

Reply:
Add the algorithm name and model method to required fields